src/Composite/CompositeParser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Pgsql\Composite;
+
+/**
+ * Composite type representation to PHP array parser for PostgreSQL Server.
+ */
+final class CompositeParser
+{
+    /**
+     * Converts composite type value from PostgreSQL to PHP array
+     *
+     * @param string $value String to convert.
+     */
+    public function parse(string $value): array|null
+    {
+        if ($value[0] !== '(') {
+            return null;
+        }
+
+        return $this->parseComposite($value);
+    }
+
+    /**
+     * Parses PostgreSQL composite type value encoded in string.
+     *
+     * @param string $value String to parse.
+     */
+    private function parseComposite(string $value): array
```

Comment:
Seems, this parser will be to work incorrect with strings contains `)` or `,` or quotes into values. For example, I change currency code in tests to `U)SD` and this fail tests.

Reply:
https://www.postgresql.org/docs/current/rowtypes.html#ROWTYPES-IO-SYNTAX
_fields containing parentheses, commas, double quotes, or backslashes must be double-quoted._

Reply:
> https://www.postgresql.org/docs/current/rowtypes.html#ROWTYPES-IO-SYNTAX
fields containing parentheses, commas, double quotes, or backslashes must be double-quoted.

DBAL should do it, right?

```php
$command->insert('test_composite_type', [
	'price_col' => ['value' => 10.0, 'currency_code' => 'US)D'],
	// ...
])->execute();

$query = (new Query($db))->from('test_composite_type')->one();

// Expected value: ['value' => 10.0, 'currency_code' => 'US)D']
// Current value: ['value' => 10.0, 'currency_code' => 'US)']
$priceColPhpType = $tableSchema->getColumn('price_col')->phpTypecast($query['price_col']);
```